src/PollySandbox/ResilienceStrategyStore.cs

```diff
@@ -0,0 +1,288 @@
+
+            return ValueTask.FromResult(TimeSpan.FromMilliseconds(delay));
+        }
+    }
+
+    private void ApplyTimeout(ResilienceStrategyBuilder builder, ApiEndpointOption endpoint)
+    {
+        var strategy = _registry.GetOrAdd($"{endpoint.Name}-Timeout", (key) => new ResilienceStrategyBuilder().AddTimeout(new TimeoutStrategyOptions()
+        {
+            OnTimeout = OnTimeout,
+            StrategyName = key,
+            Timeout = endpoint.Timeout.Add(TimeSpan.FromSeconds(1)),
+        }).Build());
+
+        builder.AddStrategy(strategy);
+    }
+
+    private ValueTask OnCircuitOpened(OutcomeArguments<object, OnCircuitOpenedArguments> arguments)
+    {
+        string operationKey = arguments.Context.OperationKey ?? "Unknown";
+
+        _metricsPublisher.Increment($"polly.circuitbreaker.open.{operationKey.ToLowerInvariant()}");
+
+        Log.CircuitOpened(
+            _logger,
+            arguments.Exception,
+            operationKey,
+            arguments.Arguments.BreakDuration);
+
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnCircuitClosed(OutcomeArguments<object, OnCircuitClosedArguments> arguments)
+    {
+        string operationKey = arguments.Context.OperationKey ?? "Unknown";
+
+        _metricsPublisher.Increment($"polly.circuitbreaker.closed.{operationKey.ToLowerInvariant()}");
+
+        Log.CircuitClosed(_logger, operationKey);
+
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnRetry(OutcomeArguments<object, OnRetryArguments> arguments)
+    {
+        string operationKey = arguments.Context.OperationKey ?? "Unknown";
+        string attempt = arguments.Arguments.Attempt.ToString(CultureInfo.InvariantCulture);
+
+        _metricsPublisher.Increment($"polly.retry.{operationKey.ToLowerInvariant()}.{attempt}");
+
+        Log.Retry(
+            _logger,
+            arguments.Exception,
+            operationKey,
+            arguments.Arguments.Attempt,
+            arguments.Arguments.RetryDelay);
+
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnTimeout(OnTimeoutArguments arguments)
+    {
+        string operationKey = arguments.Context.OperationKey ?? "Unknown";
+
+        _metricsPublisher.Increment($"polly.timeout.{operationKey.ToLowerInvariant()}");
+
+        Log.Timeout(
+            _logger,
+            arguments.Exception,
+            operationKey,
+            arguments.Timeout);
+
+        return ValueTask.CompletedTask;
+    }
+
+    private ValueTask OnRateLimited(OnRateLimiterRejectedArguments arguments)
+    {
+        string operationKey = arguments.Context.OperationKey ?? "Unknown";
+
+        _metricsPublisher.Increment($"polly.ratelimited.{operationKey.ToLowerInvariant()}");
+
+        Log.RateLimited(_logger, operationKey, arguments.RetryAfter);
+
+        return ValueTask.CompletedTask;
+    }
+
+    [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+    private static partial class Log
```

Comment:
Telemetry is now built-in into Polly V8 so this class might not be necessary anymore (alongside all the custom events you are registering to).